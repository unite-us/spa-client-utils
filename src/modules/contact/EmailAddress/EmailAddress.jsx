import React from 'react';
import PropTypes from 'prop-types';
import { truncate } from 'lodash';
import classnames from 'classnames';

const EmailAddress = ({ className, disableLink, emailAddress }) => {
  const email = emailAddress.email_address ? emailAddress.email_address : emailAddress;

  return (
    <span className={classnames('ui-email-address', className)}>
      { disableLink ?
        email :
        <a
          href={`mailto:${email}`}
          title={email}
        >
          {truncate(email, { length: 40 })}
        </a>
      }
    </span>
  );
};

EmailAddress.propTypes = {
  /** className */
  className: PropTypes.string,
  /** true to disable the mailto link */
  disableLink: PropTypes.bool,
  /** email address string or object */
  emailAddress: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.string,
  ]).isRequired,
};

EmailAddress.defaultProps = {
  className: '',
  disableLink: false,
  isInNetworkGroup: true,
};

export default EmailAddress;
