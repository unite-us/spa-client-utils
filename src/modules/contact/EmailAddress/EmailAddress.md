```jsx
const emailAddress = {
  email_address: 'emailaddress@internet.com',
  isPrimary: true,
};

<EmailAddress emailAddress="emailaddress@internet.com" />
```
