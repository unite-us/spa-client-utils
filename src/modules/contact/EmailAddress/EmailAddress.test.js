import React from 'react';
import renderer from 'react-test-renderer';
import EmailAddress from './EmailAddress';

const emailAddress = 'email@address.com';

describe('EmailAddress', () => {
  it('renders email address as mailto link', () => {
    expect(renderer.create(
      <EmailAddress emailAddress={emailAddress} />,
    )).toMatchSnapshot();
  });

  it('renders email address as plain text', () => {
    expect(renderer.create(
      <EmailAddress disableLink emailAddress={emailAddress} />,
    )).toMatchSnapshot();
  });
});
