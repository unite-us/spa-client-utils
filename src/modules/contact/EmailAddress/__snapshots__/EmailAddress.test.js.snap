// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailAddress renders email address as mailto link 1`] = `
<span
  className="ui-email-address"
>
  <a
    href="mailto:email@address.com"
    title="email@address.com"
  >
    email@address.com
  </a>
</span>
`;

exports[`EmailAddress renders email address as plain text 1`] = `
<span
  className="ui-email-address"
>
  email@address.com
</span>
`;
